templates/palettes.html

```diff
@@ -1,7 +1,7 @@
 <div id="paletteList">
     <div data-bind="foreach: palettes">
         <div class="card">
-            <div class="card-header" data-toggle="collapse" aria-expanded="true" data-bind="attr:{id:'palette' + $index(), 'data-target':'#collapse' + $index(), 'aria-controls':'collapse' + $index()}">
+            <div class="card-header" data-toggle="collapse" aria-expanded="true" data-bind="attr:{id:'palette' + $index(), 'data-target':'#collapse' + $index(), 'aria-controls':'collapse' + $index()}, event: {click: $root.spinCollapseIcon}">
```

Comment:
You've used the "event" binding here, which is fine. There is also a binding in knockoutJS called "click", specifically for the click event. So you could write this as data-bind="click: $root.spinCollapseIcon", instead of data-bind="event: {click: $root.spinCollapseIcon}".

But this comment is just to remind you , no need to change the code.

Reply:
I remember having it as "click: $root.spinCollapseIcon" initially but it triggered the function on creation of the element, meaning each time you selected or created a node it would trigger the function and throw an error.

Reply:
I have troubles with this issue too, quite frequently.

In my experience, if you use "click: $root.spinCollapseIcon()" (note the trailing brackets), then the function will be executed at the time of binding.
If you use "click: $root.spinCollapseIcon" (no trailing brackets), then the function will be executed when the click event occurs.

Having said that, you'll see instances throughout the code where I have wrapped the function call in an anonymous function. Something like: "click: function(){$root.spinCollapseIcon();}" just to ensure the function is executed at event time (or if the function needs some arguments).